polars/polars-lazy/polars-plan/src/dsl/dt.rs

```diff
@@ -71,6 +82,10 @@ impl DateLikeNameSpace {
             move |s| match s.dtype() {
                 DataType::Datetime(_, _) => {
                     let mut ca = s.datetime().unwrap().clone();
+                    match &tz {
+                        None => (),
+                        Some(tz) => validate_time_zone(tz.to_string()),
+                    };
                     ca.set_time_zone(tz.clone());
```

Comment:
I don't think the validation should be here. But rather in the most inner `set_time_zone`. This will ensure the validation is done everywhere.